disdat/api.py

```diff
@@ -70,93 +70,138 @@ def set_aws_profile(aws_profile):
-        Or use a temp dir and copy things into the bundle when you're done.
-        If #2 then, it's easy to use a bundle object and write it to multiple
-        contexts.   We should close or destroy a bundle in case 2.
+        Default name:  If you don't provide a name, Disdat tries to use the basename in `code_ref`.
+        Default processing_name:   If you don't provide a processing name, Disdat will use a default that
+        takes into consideration your bundles upstream inputs, parameters, and code reference.
 
         Args:
-            local_context (str): Where this bundle will be output or where it was sourced from.
-            name (str): Human name for this bundle
+            local_context (Union[str, `disdat.data_context.DataContext`): The local context name or context object
+            name (str): Human name for this bundle.
+            data (union(pandas.DataFrame, tuple, None, list, dict)):  The data this bundle contains.
+            processing_name (str):  A name that indicates a bundle was made in an identical fashion.
+            owner (str):  The owner of the bundle.  Default getpass.getuser()
+            tags (dict):  (str,str) dictionary of arbitrary user tags.
+            params (dict(str:str)): Dictionary of parameters that <code_method> used to produce this output.
+            dependencies (dict(str:bundle)):  Dictionary of argname: bundle, Bundles used to produce this output.
+            code_method (str):  A reference to code that created this bundle. Default None
+            vc_info (tuple):  Version control information triple: e.g. tuple(git_repo , git_commit, branch)
+            start_time (float):  Start time of the process that produced the bundle. Default time.now()
+            stop_time (float):   Stop time of the process that produced the bundle. Default time.now()
+
         """
 
-        self.fs = _get_fs()
+        self._fs = _get_fs()
 
         try:
-            self.data_context = self.fs.get_context(local_context)
+            if isinstance(local_context, DataContext):
+                self.data_context = local_context
+            elif isinstance(local_context, str):
+                self.data_context = self._fs.get_context(local_context)
+            else:
+                raise Exception("Unable to create Bundle because no context found with name[{}]".format(local_context))
         except Exception as e:
             _logger.error("Unable to allocate bundle in context: {} ".format(local_context, e))
             return
 
-        super(Bundle, self).__init__(human_name=name, owner=owner)
+        self._local_dir = None
+        self._remote_dir = None
+        self._closed = False     # Bundle is closed and immutable
+        self._data = None        # The df, array, dictionary the user wants to store
+
+        super(Bundle, self).__init__(human_name=name, #'' if name is None else name,
+                                     owner='unknown' if owner is None else owner,
```

Comment:
in punch I use getpass.getuser() for the default owner

Reply:
I do that in other places, so I should make that consistent.   Done.